Add dates to Calendar

hawkrives opened the pull request:
Closes #1232
Closes #1233 

Kind | Before | After
--- | --- | ---
All Day | <img width="375" alt="screen shot 2017-08-05 at 11 22 55 pm" src="https://user-images.githubusercontent.com/464441/29000585-09930124-7a35-11e7-99aa-6ffc5afa1f02.png"> | <img width="375" alt="screen shot 2017-08-05 at 11 22 31 pm" src="https://user-images.githubusercontent.com/464441/29000583-0989c4ec-7a35-11e7-85e9-e3df160068b4.png">
Normal | <img width="375" alt="screen shot 2017-08-05 at 11 22 49 pm" src="https://user-images.githubusercontent.com/464441/29000584-098ffba0-7a35-11e7-8e36-32e4ce6798ed.png"> | <img width="375" alt="screen shot 2017-08-05 at 11 22 07 pm" src="https://user-images.githubusercontent.com/464441/29000582-0981c878-7a35-11e7-9973-29826d1a6d02.png">

I also removed some odd duplication between the iOS and Android views.

hawkrives committed:
clean up the iOS event detail view to use the shared functions

hawkrives committed:
add dates to calendar detail views

drewvolz commented on source/views/calendar/times.js:
```diff
@@ -41,3 +41,45 @@ export function times(event: EventType) {
+      : 'MMM. D h:mm A'
+    end = `${event.endTime.format(endFormat)}`
+  } else if (sillyZeroLength) {
+    start = `Starts at ${startTimeFormatted}`
```
`startTimeFormatted` will have a date in this case.

I'm going to suggest changing the word "on" for "at".
> **Current** "Starts `at` Aug. 13 12:00 PM"
> **Proposed**: "Starts `on` Aug. 13 12:00 PM"

hawkrives replied:
good eye

drewvolz reviewed (approved):
👍 :shipit:

drewvolz committed:
prettier whitespace

hawkrives committed:
use drew's suggestion

`startTimeFormatted` will have a date in this case.

I'm going to suggest changing the word "on" for "at".

> **Current** "Starts `at` Aug. 13 12:00 PM"
> **Proposed**: "Starts `on` Aug. 13 12:00 PM"
